// unstructured-client/dist/esm/lib/config.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { pathToFunc } from "./url.js";
/**
 * Serverless SaaS API
 */
export const ServerSaasApi = "saas-api";
/**
 * Hosted API Free
 */
export const ServerFreeApi = "free-api";
/**
 * Development server
 */
export const ServerDevelopment = "development";
/**
 * Contains the list of servers available to the SDK
 */
export const ServerList = {
    [ServerSaasApi]: "https://api.unstructuredapp.io",
    [ServerFreeApi]: "https://api.unstructured.io",
    [ServerDevelopment]: "http://localhost:8000",
};
export function serverURLFromOptions(options) {
    let serverURL = options.serverURL;
    const params = {};
    if (!serverURL) {
        const server = options.server ?? ServerSaasApi;
        serverURL = ServerList[server] || "";
    }
    const u = pathToFunc(serverURL)(params);
    return new URL(u);
}
export const SDK_METADATA = {
    language: "typescript",
    openapiDocVersion: "1.0.81",
    sdkVersion: "0.22.0",
    genVersion: "2.548.6",
    userAgent: "speakeasy-sdk/typescript 0.22.0 2.548.6 1.0.81 unstructured-client",
};
